lib/str.js

```diff
@@ -24,19 +24,25 @@ const Str = {
     },
 
     /**
-     * Converts a USD string into th number of cents it represents.
+     * Converts a currency string into the number of cents it represents.
      *
-     * @param {String}  amountStr     A string representing a USD value.
-     * @param {Boolean} allowFraction Flag indicating if fractions of cents should be
-     *                               allowed in the output.
+     * @param {String}  amountStr     String representing a currency symbol and value, like $4.02 or Bs.S97.9
+     * @param {Boolean} allowFraction Flag indicating if fractions of cents should be allowed in the output.
      *
      * @return {Number} The cent value of the @p amountStr.
      */
-    fromUSDToNumber(amountStr, allowFraction) {
-        let amount = String(amountStr).replace(/[^\d.\-()]+/g, '');
-        if (amount.match(/\(.*\)/)) {
-            const modifiedAmount = amount.replace(/[()]/g, '');
-            amount = `-${modifiedAmount}`;
+    fromCurrencyToNumber(amountStr, allowFraction) {
```

Comment:
It looks like changes introduced in this PR have bough this regression https://github.com/Expensify/Expensify/issues/309211

1. If the amount field is not present in the CSV file, it throws an error, originally we used to default to 0.
2. If the amount is negative the amount calculated is `NaN` and further defaulting to 0?

<img width="902" alt="Screenshot 2023-08-21 at 4 41 23 PM" src="https://github.com/Expensify/expensify-common/assets/35863227/800dc2eb-e3f9-410d-9b89-ce9955d85fe0">
<img width="909" alt="Screenshot 2023-08-21 at 4 42 56 PM" src="https://github.com/Expensify/expensify-common/assets/35863227/d4332e0d-b129-4dee-889e-45879d3a7d29">
<img width="1012" alt="Screenshot 2023-08-21 at 4 58 26 PM" src="https://github.com/Expensify/expensify-common/assets/35863227/8040d3d3-ae82-4178-922b-7c7d83319551">
<img width="1113" alt="Screenshot 2023-08-21 at 4 58 46 PM" src="https://github.com/Expensify/expensify-common/assets/35863227/3bb02ec7-1a6f-4786-be3e-40894bfdbc21">

Reply:
This almost seems to cause another regression here: https://github.com/Expensify/Expensify/issues/309227

Reply:
Can you have a negative per diem? Seems like defaulting to 0 in that case would be correct, right?

For the other, maybe something like changing [this line](https://github.com/Expensify/expensify-common/pull/561/files?diff=unified#diff-eb3aad4f6d64ed33948e40b58a65d583ddf513521d7a3e23871e5f275ac907fcR47) to have a default case of 0?

Will defer to @Li357 though